### ISSUE TITLE
Identify and label WorkloadEntry graph nodes

### ISSUE THREAD

**Issue by nrfox** [kiali/kiali#4223]
In a similar manner to `ServiceEntry` nodes being identified and labeled with `SEInfo`, `WorkloadEntry` nodes should be identified and labeled to distinguish them from regular workload nodes. This will allow the graph to visually distinguish these workloads from pod-based workloads in some way or to change other node related behavior such as providing links to the Istio Config for the `WorkloadEntry` instead of linking to the (non-existent for `WorkloadEntry`) workload details page.

This will most likely require correlating `WorkloadEntry` objects with telemetry data. [Standard Istio metrics](https://istio.io/latest/docs/reference/config/metrics/) are available from the VM-based workloads.

**Comment by jshaughn** [comment]
@nrfox Assigning you to this one so we can continue to improve the WorkloadEntry experience.  As we've discussed off-issue, it's not totally clear how to actually identify a WE from the telemetry, as there isn't anything we've seen that would distinguish the source_workload or destination_workload as being a mesh_internal VM as opposed to a kubernetes/pod workload.  One possibility is to just assume that if we can't find it in the workloads queried from k8s that it must be a WE/VM.

As for how to represent it in the graph, I'm sort of loathe to add a new shape.  And really, they are pretty much still workloads.  Maybe a new badge is sufficient.  Or maybe a slightly different look, like a double-circle or something.  I don't know.

**Comment by jshaughn** [comment]
@nrfox No, I think he's referring to the istiod service registry, which can include entries that are istio-specific, past what is defined in k8s.  We should take a look at this to see if workload entries can be easily identified.  I haven'tplayed with querying that registry directly, @lucasponce may be able to point to an example.

**Comment by nrfox** [comment]
> @nrfox No, I think he's referring to the istiod service registry, which can include entries that are istio-specific, past what is defined in k8s. We should take a look at this to see if workload entries can be easily identified. I haven'tplayed with querying that registry directly, @lucasponce may be able to point to an example.

Ah I seem to remember trying this in the past but I don't recall what the mechanics are of querying that registry or if it's easily accessible. Would querying the registry instead of getting that info from kube just be used for cases where there is no clear way to infer the information from the kube objects?

**Comment by lucasponce** [comment]
@nrfox Istio provides a registry that is used in Kiali for several features, as you pointed, it's the real "networking" table used to propagated the endpoints to the proxies.

It's already added to the business layer, the `RegistryStatusService` from the layer will give you this data, and it's already cached.

At the moment Kiali query the "/debug/syncz" and "/debug/registryz" endpoints that gives all info about external routes as well:

For example, for a ServiceEntry:

```
{
  "Attributes": {
    "ServiceRegistry": "External",
    "Name": "edition.cnn.com",
    "Namespace": "default",
    "Labels": null,
    "UID": "",
    "ExportTo": null,
    "LabelSelectors": null,
    "ClusterExternalAddresses": null,
    "ClusterExternalPorts": null
  },
  "ports": [
    {
      "name": "http-port",
      "port": 80,
      "protocol": "HTTP"
    },
    {
      "name": "https",
      "port": 443,
      "protocol": "HTTPS"
    }
  ],
  "creationTime": "2021-08-10T10:10:01Z",
  "hostname": "edition.cnn.com",
  "address": "0.0.0.0",
  "autoAllocatedAddress": "240.240.0.1",
  "Mutex": {},
  "Resolution": 1,
  "MeshExternal": true
},
```

The WorkloadEntries should have a similar entry that you could use, I don't have an example with VM but you can query the registry locally using:

```
export ISTIOD_POD=$(kubectl get pod -l app=istiod -n istio-system -o jsonpath={.items..metadata.name})
kubectl exec "$ISTIOD_POD" -n istio-system -- curl http://localhost:15014/debug/registryz
```

This other endpoint fetches more details about the endpoints, but it's not yet used in Kiali:

```
export ISTIOD_POD=$(kubectl get pod -l app=istiod -n istio-system -o jsonpath={.items..metadata.name})
kubectl exec "$ISTIOD_POD" -n istio-system -- curl http://localhost:15014/debug/endpointz
```

So, if useful, it can be used to correlate other information and to identify easily this workloads.

It's still an open topic, these endpoints are used for validations as well and in the future we may want to adjust some logic to replace other use cases, but that's more a off-topic discussion (just giving visibility that it's an active area).

**Comment by nrfox** [comment]
I opened up a draft PR: #4284 which just looks at the app/version labels on each node and sees if there's a corresponding app/version label on a `WorkloadEntry` in the same namespace. That may be too simplistic or narrow and can be updated to look at a superset of the labels if need be. In this case, I'm not sure we need to look at the istio debug info to correlate the `WorkloadEntry` istio config objects with graph Workload nodes unless correlating the two with the istio debug info is preferred?

**Comment by lucasponce** [comment]
> I opened up a draft PR: #4284 which just looks at the app/version labels on each node and sees if there's a corresponding app/version label on a WorkloadEntry in the same namespace. 

I think this is a good starting point.

Perhaps I introduced some noise when I proposed to use the registry. 
There would be cases where the Istio definition is added directly into the registry, then using the registry may help (that's happening in some validations under federation scenarios).

But linking the "WorkloadEntry" Istio config from an appender sounds good to me.
